code/modules/food_and_drinks/food/snacks_vend.dm

```diff
@@ -178,4 +178,17 @@
 	if(!opened)
 		to_chat(user, span_warning("[src]'s lid hasn't been opened!"))
 		return FALSE
-	return ..()
\ No newline at end of file
+	return ..()
+
+/obj/item/reagent_containers/food/snacks/vermin
+	name = "vermin bites"
+	desc = "A small can with a cartoon mouse on the label. A noise that sounds suspiciously like squeaking can be heard coming from inside."
+	icon_state = "verminbites"
+	tastes = list("rats" = 1 , "mouse" = 2, "cheese" = 1)
+	foodtype = MEAT
+
+/obj/item/reagent_containers/food/snacks/vermin/attack_self(mob/user)
+	. = ..()
+	to_chat(user, span_warning("You crack open [src], a mouse falling out from inside!"))
+	new /mob/living/simple_animal/mouse(get_turf(src))
+	qdel(src)
```

Comment:
```suggestion
/obj/item/reagent_containers/food/snacks/vermin
	name = "vermin bites"
	desc = "A small can with a cartoon mouse on the label. A noise that sounds suspiciously like squeaking can be heard coming from inside."
	icon_state = "verminbites"
	tastes = list("rats" = 1 , "mouse" = 2, "cheese" = 1)
	foodtype = MEAT
	/// What animal does the snack contain?
	var/contained_animal = /mob/living/simple_animal/mouse

/obj/item/reagent_containers/food/snacks/vermin/attack_self(mob/user)
	. = ..()
	to_chat(user, span_warning("You crack open [src], [contained_animal] falling out from inside!"))
	new contained_animal(get_turf(src))
	qdel(src)
```
Fair enough. This should work if it's something you're interested in adding. I haven't tested it though, so you'd need to do that. Or feel free not to add it, just a suggestion.